cmd/server.go

```diff
@@ -92,6 +92,11 @@ func serverRun() {
 		ctx.Redirect(http.StatusFound, "/maiden")
 	})
 
+	// defer routing to web app
+	r.NoRoute(func(ctx *gin.Context) {
```

Comment:
If I'm understanding the `NoRoute` directive it is basically overriding the default 404 response and instead returning the application index page. I'm concerned this will hamper debugging and potentially trigger some confusion elsewhere in code - for example if I have `http://norns.local/blah/balh/blah` the backend responds with the main index page (for the maiden app) to being loaded *but* with the bad URL as the location.

One pattern which I've seen in single page apps is that app side URL routing uses fragment syntax which makes it more clear which part of the stack is doing the routing; for example `http://norns.local/maiden#<stuff>`

Reply:
Yes, exactly. I think the fragments are a much better idea, as long as they don't break back/forward in history.

Reply:
Removed server/go changes.